lib/resty/core/base.lua

```diff
@@ -5,6 +5,7 @@ local ffi = require 'ffi'
 local ffi_new = ffi.new
 local error = error
 local ceil = math.ceil
+local subsystem = ngx.config.subsystem
```

Comment:
Merely a suggestion, hoping it is not intrusive: since this is being called before running the version ngx_lua check (for obvious and valid reasons), the error will be somewhat hard to decipher for users if this module is being used with an older ngx_lua module. Should we fool-proof it (like we fool-proof the ngx_lua version below) with something like:

```lua
local subsystem
do
  local ok = pcall(function() subsystem = ngx.config.subsystem end)
  if not ok then
    -- 0.10.10 was the first version to support ngx.config.subsystem, but this module now needs 0.10.11 anyways
    error("ngx_lua 0.10.11+ required")
  end
end
```

I haven't tested this.

Reply:
Actually, this might be enough, `config` will not throw an error on an `nil` key iirc:
```lua
if not subsystem then
  error("ngx_lua 0.10.11+ required")
end
```

Reply:
(The below test is even more conservative in that it is even testing for the existence of `ngx.config`)